elasticai_api/common/data_shard_service.py

```diff
@@ -34,12 +34,13 @@ def get_current_task(self):
         return self._current_task
 
     def get_task(self, task_type=None):
-        task = self._mc.get_task(task_type)
-        if task.type == elasticai_api_pb2.TRAINING:
-            self._pending_tasks.append(task)
-            if len(self._pending_tasks) == 1:
-                self._current_task = task
-        return task
+        with self._lock:
+            task = self._mc.get_task(task_type)
+            if task.type == elasticai_api_pb2.TRAINING:
+                self._pending_tasks.append(task)
```

Comment:
The lock can be added before `self._pending_tasks.append(task)`

Reply:
Good suggestion and fixed it.